web/src/settings_preferences.ts

```diff
@@ -152,23 +156,25 @@ function user_default_language_modal_post_render(): void {
                 setting_value,
             );
 
-            change_display_setting(
+            const $spinner = $("#settings_content").find(".general-settings-status").expectOne();
+            $spinner.fadeTo(0, 1);
+            loading.make_indicator($spinner, {text: settings_ui.strings.saving});
+            channel.patch({
+                url: "/json/settings",
                 data,
-                $("#settings_content").find(".general-settings-status"),
-                undefined,
-                undefined,
-                $t_html(
-                    {
-                        defaultMessage:
-                            "Saved. Please <z-link>reload</z-link> for the change to take effect.",
-                    },
-                    {
-                        "z-link": (content_html) =>
-                            `<a class='reload_link'>${content_html.join("")}</a>`,
-                    },
-                ),
-                true,
-            );
+                success(response_data) {
+                    const appear_after = 500;
+                    const data = translated_message_response_schema.parse(response_data);
```

Comment:
This can directly be done as `z.object({msg: z.string()}).parse(response_data)`. I don't think there is any need of defining the schema in `translated_message_response_schema` variable.